pkg/infra/baseagent.go

```diff
@@ -213,6 +215,29 @@ func (a *Base) HealthInfo() *health.HealthInfo {
 	return a.tower.healthInfo
 }
 
+func (a *Base) ActiveTreaties() map[uuid.UUID]messages.Treaty {
+	return a.activeTreaties
+}
+
+func (a *Base) AddToActiveTreaties(treaty messages.Treaty) {
+	a.activeTreaties[treaty.ID()] = treaty
+}
+
+func (a *Base) DeleteFromActiveTreaties(treaty messages.Treaty) {
+	delete(a.activeTreaties, treaty.ID())
+}
+
+func (a *Base) RespondToTreaty(msg messages.ProposeTreatyMessage, senderFloor int, response bool) messages.TreatyResponseMessage {
+	treatyToReview := msg.Treaty()
+	if response {
+		treatyToReview.SignTreaty()
+		a.AddToActiveTreaties(treatyToReview)
+	}
+	reply := msg.Reply(a.ID(), senderFloor, response)
+
+	return reply
+}
```

Comment:
I'm not sure this function is necessary - this would be what you do to in `HandleProposeTreaty()` I think